os.kieserver.launch/added/kieserver-setup.sh

```diff
@@ -31,25 +31,80 @@ function filterKieJmsFile() {
+
     quartzPropFile="${1}"
     if [ -e ${quartzPropFile} ] ; then
-        if [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.MySQL"* ]]; then
-            sed -i "s,org.quartz.jobStore.driverDelegateClass=,org.quartz.jobStore.driverDelegateClass=org.quartz.impl.jdbcjobstore.StdJDBCDelegate," ${quartzPropFile}
-            quartzDriverDelegateSet="true"
-        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.PostgreSQL"* ]]; then
-            sed -i "s,org.quartz.jobStore.driverDelegateClass=,org.quartz.jobStore.driverDelegateClass=org.quartz.impl.jdbcjobstore.PostgreSQLDelegate," ${quartzPropFile}
-            quartzDriverDelegateSet="true"
+
+        # get the first db configuration using legacy datasource configuration:
+        if [ "x${DB_SERVICE_PREFIX_MAPPING}" != "x" ]; then
+            local serviceMappingName=${DB_SERVICE_PREFIX_MAPPING%%,*}
+            local prefix=${serviceMappingName#*=}
+            DEFAULT_JNDI=$(find_env "${prefix}_JNDI")
         fi
-        if [ "x${DB_JNDI}" != "x" ]; then
-            sed -i "s,org.quartz.dataSource.managedDS.jndiURL=,org.quartz.dataSource.managedDS.jndiURL=${DB_JNDI}," ${quartzPropFile}
+        # get the first db configuration using DATASOURCES env:
+        if [ "x${DATASOURCES}" != "x" ]; then
+            local prefix="${DATASOURCES%%,*}"
+            DEFAULT_JNDI=$(find_env "${prefix}_JNDI")
+        fi
+
+        if [ "x${DEFAULT_JNDI}" != "x" ]; then
             quartzManagedJndiSet="true"
         fi
         if [ "x${QUARTZ_JNDI}" != "x" ]; then
-            sed -i "s,org.quartz.dataSource.notManagedDS.jndiURL=,org.quartz.dataSource.notManagedDS.jndiURL=${QUARTZ_JNDI}," ${quartzPropFile}
             quartzNotManagedJndiSet="true"
         fi
+
+        if [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.MySQL"* ]]; then
+            quartzDriverDelegateSet="true"
+            DB_TYPE="MYSQL"
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.PostgreSQL"* ]]; then
+            QUARTZ_DELEGATE_CLASS="org.quartz.impl.jdbcjobstore.PostgreSQLDelegate"
+            quartzDriverDelegateSet="true"
+            DB_TYPE="POSTGRESQL"
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == *"Oracle"* ]]; then
+            log_warning "For Oracle 10G use the org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect to avoid frequent Hibernate warning HHH000444 message in the logs."
+            QUARTZ_DELEGATE_CLASS="org.quartz.impl.jdbcjobstore.oracle.OracleDelegate"
+            # Oracle classes needs to be accessible by the quartz, in this case oracle driver needs to be added on jboss-deployment-structure.xml file
+            if [ "x${QUARTZ_DRIVER}" != "x" ]; then
+                sed -i "s|<\!--EXTRA_DEPENDENCY-->|<module name=\"deployment.${QUARTZ_DRIVER}\"/>\n      <\!--EXTRA_DEPENDENCY-->|" $JBOSS_HOME/standalone/deployments/kie-server.war/WEB-INF/jboss-deployment-structure.xml
```

Comment:
This expects the driver to be installed as a deployment,  however, the driver could also be installed as a module
e.g., as described here

https://blog.openshift.com/conntecting-to-an-external-oracle-database/

it would perhaps be better to explicitly ask for a "QUARTZ_DRIVER_MODULE" , to let the user provide anything, not necessarily with "deployment." prefix (and describe it in the documentation)

Reply:
@spolti Can you please address @maschmid 's concern above?

Reply:
@maschmid I agree with you, thanks, and sry for the delay, I complete forgot about it =/
@errantepiphany sure thing.